packages/ember/addon/instance-initializers/sentry-performance.ts

```diff
@@ -390,57 +390,56 @@ export async function instrumentForPerformance(appInstance: ApplicationInstance)
-
-        if (routerService.externalRouter) {
-          // Using ember-engines-router-service in an engine.
-          routerService = routerService.externalRouter;
-        }
-        if (routerService._hasMountedSentryPerformanceRouting) {
-          // Routing listens to route changes on the main router, and should not be initialized multiple times per page.
-          return;
-        }
-        if (!routerService.recognize) {
-          // Router is missing critical functionality to limit cardinality of the transaction names.
-          return;
-        }
-        routerService._hasMountedSentryPerformanceRouting = true;
-        _instrumentEmberRouter(routerService, routerMain, config, customStartTransaction, startTransactionOnPageLoad);
-      },
-      idleTimeout,
-      ...browserTracingOptions,
-    }),
-  ];
-
-  class FakeBrowserTracingClass {
-    static id = 'BrowserTracing';
-    public name = FakeBrowserTracingClass.id;
-    setupOnce() {
-      // noop - We're just faking  this class for a lookup
+  const browserTracing = new BrowserTracing({
+    routingInstrumentation: (customStartTransaction, startTransactionOnPageLoad) => {
+      const routerMain = appInstance.lookup('router:main');
+      let routerService = appInstance.lookup('service:router') as
+        | RouterService & { externalRouter?: RouterService; _hasMountedSentryPerformanceRouting?: boolean };
+
+      if (routerService.externalRouter) {
+        // Using ember-engines-router-service in an engine.
+        routerService = routerService.externalRouter;
+      }
+      if (routerService._hasMountedSentryPerformanceRouting) {
+        // Routing listens to route changes on the main router, and should not be initialized multiple times per page.
+        return;
+      }
+      if (!routerService.recognize) {
+        // Router is missing critical functionality to limit cardinality of the transaction names.
+        return;
+      }
+      routerService._hasMountedSentryPerformanceRouting = true;
+      _instrumentEmberRouter(routerService, routerMain, config, customStartTransaction, startTransactionOnPageLoad);
+    },
+    idleTimeout,
+    ...browserTracingOptions,
+  });
+
+  if (macroCondition(isTesting())) {
+    class FakeBrowserTracingClass {
+      static id = 'BrowserTracing';
+      public name = FakeBrowserTracingClass.id;
+      setupOnce() {
+        // noop - We're just faking  this class for a lookup
+      }
     }
-  }
 
-  if (
-    isTesting() &&
-    Sentry.getCurrentHub()?.getIntegration(
-      // This is a temporary hack because the BrowserTracing integration cannot have a static `id` field for tree
-      // shaking reasons. However, `getIntegration` needs that field.
-      FakeBrowserTracingClass,
-    )
-  ) {
-    // Initializers are called more than once in tests, causing the integrations to not be setup correctly.
-    return;
+    if (
+      Sentry.getCurrentHub()?.getIntegration(
```

Comment:
l (and out of scope, feel free to ignore): Any reason why we couldn't use `BaseClient::getIntegrationbyId` instead? Then we probably wouldn't need this fake class, right?

Reply:
Sadly `BrowserTracing` doesn't have an ID, right now 🙈

Reply:
It doesn't need one I think. At least that's how I'm accessing the browsertracing integration in #7626:

https://github.com/getsentry/sentry-javascript/blob/df10ac4582f695e021df403071f4d015c4570365/packages/sveltekit/src/client/load.ts/#L110-L113

Reply:
Ah nice, it works because it actually checks the `name` field 👍